src/v/transform/api.cc

```diff
@@ -446,6 +450,23 @@ ss::future<> service::start() {
     co_await _batcher->start();
     co_await _manager->start();
     register_notifications();
+
+    _log_manager = std::make_unique<logging::manager<ss::lowres_clock>>(
+      _self,
+      std::make_unique<logging::rpc_client>(
+        &_rpc_client->local(), &_metadata_cache->local()),
+      config::shard_local_cfg().data_transforms_logging_buffer_capacity_bytes(),
+      config::shard_local_cfg().data_transforms_logging_line_max_bytes.bind(),
+      config::shard_local_cfg()
+        .data_transforms_logging_flush_interval_ms.bind());
+    auto fut = co_await ss::coroutine::as_future(_log_manager->start());
+    if (fut.failed()) {
+        vlog(
+          tlog.error,
+          "Failed to start transform::logging::manager: {}",
+          fut.get_exception());
+    }
+    co_return co_await std::move(fut);
```

Comment:
I think this will probably crash or hang. Once you eat the exception out of the future, it's put into an invalid state. When you co_await on it, at least from my reading of coroutine.hh, seastar will wait for the future to become available, but that will never happen because it _was_ available (before get_exception) but now it is marked invalid.

Reply:
Ack

Reply:
Can we just remove the as future thing here. I dont think it adds value

Reply:
should be something like this, right?

```suggestion
    if (fut.failed()) {
        vlog(
          tlog.error,
          "Failed to start transform::logging::manager: {}",
          fut.get_exception());
        co_return;
    }
    co_return co_await std::move(fut);
```

Reply:
@rockwotj - oh yeah, that's fine. will do later

Reply:
What about just

```
co_await _log_manager->start();
```

I don't think we want to silently fail to start the manager.

Reply:
Yeah, that's what I'm committing right now.